NPE with new bugfix release

According to Play Console this happens with the new version 2014.11.03:

```
java.lang.NullPointerException
at java.text.SimpleDateFormat.validatePattern(SimpleDateFormat.java:267)
at java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:356)
at cgeo.geocaching.connector.gc.GCLogin.parseGcCustomDate(GCLogin.java:283)
at cgeo.geocaching.connector.gc.GCLogin.parseGcCustomDate(GCLogin.java:287)
at cgeo.geocaching.connector.gc.GCParser.parseSearch(GCParser.java:213)
at cgeo.geocaching.connector.gc.GCParser.searchByAny(GCParser.java:860)
at cgeo.geocaching.connector.gc.GCParser.searchByCoords(GCParser.java:875)
at cgeo.geocaching.connector.gc.GCMap.searchByViewport(GCMap.java:354)
at cgeo.geocaching.connector.gc.GCMap.searchByViewport(GCMap.java:248)
at cgeo.geocaching.connector.gc.GCConnector.searchByViewport(GCConnector.java:182)
at cgeo.geocaching.connector.ConnectorFactory$1.call(ConnectorFactory.java:193)
at cgeo.geocaching.connector.ConnectorFactory$1.call(ConnectorFactory.java:190)
at cgeo.geocaching.SearchResult$3$1.call(SearchResult.java:323)
at cgeo.geocaching.SearchResult$3$1.call(SearchResult.java:320)
at rx.internal.operators.OperatorMap$1.onNext(OperatorMap.java:54)
at rx.internal.operators.NotificationLite.accept(NotificationLite.java:147)
at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:141)
at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.access$000(OperatorObserveOn.java:61)
at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$1.call(OperatorObserveOn.java:123)
at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:841)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: GCConnector.class
at rx.internal.operators.OperatorMap$1.onNext(OperatorMap.java:56)
... 9 more
```

User message:
- Crash when populating the map
- Crash when accessing the map

--- a/main/src/cgeo/geocaching/connector/gc/GCConstants.java
+++ b/main/src/cgeo/geocaching/connector/gc/GCConstants.java
@@ -24,6 +24,8 @@ public final class GCConstants {
     public final static @NonNull String URL_MAP_INFO = GC_TILE_URL + "map.info";
     /** Tile itself */
     public final static @NonNull String URL_MAP_TILE = GC_TILE_URL + "map.png";
+    /** Format used by geocaching.com when user is not logged in. */
+    public static final String DEFAULT_GC_DATE = "MM/dd/yyyy";
 
     /**
      * Patterns for parsing the result of a (detailed) search

--- a/main/src/cgeo/geocaching/connector/gc/GCLogin.java
+++ b/main/src/cgeo/geocaching/connector/gc/GCLogin.java
@@ -50,7 +50,7 @@ public class GCLogin extends AbstractLogin {
     }
 
     private static StatusCode resetGcCustomDate(final StatusCode statusCode) {
-        Settings.setGcCustomDate("MM/dd/yyyy");
+        Settings.setGcCustomDate(GCConstants.DEFAULT_GC_DATE);
         return statusCode;
     }
 

--- a/main/src/cgeo/geocaching/settings/Settings.java
+++ b/main/src/cgeo/geocaching/settings/Settings.java
@@ -159,7 +159,7 @@ public class Settings {
             e.putBoolean(getKey(R.string.pref_log_offline), old.getBoolean(getKey(R.string.pref_log_offline), false));
             e.putBoolean(getKey(R.string.pref_choose_list), old.getBoolean(getKey(R.string.pref_choose_list), true));
             e.putBoolean(getKey(R.string.pref_loaddirectionimg), old.getBoolean(getKey(R.string.pref_loaddirectionimg), true));
-            e.putString(getKey(R.string.pref_gccustomdate), old.getString(getKey(R.string.pref_gccustomdate), null));
+            e.putString(getKey(R.string.pref_gccustomdate), old.getString(getKey(R.string.pref_gccustomdate), GCConstants.DEFAULT_GC_DATE));
             e.putInt(getKey(R.string.pref_showwaypointsthreshold), old.getInt(getKey(R.string.pref_showwaypointsthreshold), SHOW_WP_THRESHOLD_DEFAULT));
             e.putString(getKey(R.string.pref_cookiestore), old.getString(getKey(R.string.pref_cookiestore), null));
             e.putBoolean(getKey(R.string.pref_opendetailslastpage), old.getBoolean(getKey(R.string.pref_opendetailslastpage), false));
@@ -507,7 +507,9 @@ public class Settings {
      * @return User selected date format on GC.com
      */
     public static String getGcCustomDate() {
-        return getString(R.string.pref_gccustomdate, null);
+        // We might have some users whose stored value is null, which is invalid. In this case, we use the default.
+        return StringUtils.defaultString(getString(R.string.pref_gccustomdate, GCConstants.DEFAULT_GC_DATE),
+                GCConstants.DEFAULT_GC_DATE);
     }
 
     public static boolean isExcludeMyCaches() {

